src/components/MapboxMap.tsx

```diff
@@ -66,6 +67,11 @@ export interface MapboxMapProps<T> {
   onDrag?: OnDragHandler
 }
 
+/**
+ * Defined mapbox options as defaults. These options are also needed for Mapbox Static Images API.
+ */
+type DefaultMapboxOptions = Required<Pick<MapboxOptions, 'center' | 'style' | 'zoom'>>;
```

Comment:
The original intention was for the three fields here ('center' | 'style' | 'zoom') to always be defined as part of our default options so we can construct a valid mapbox static image url. This would help bypass the need for manual checking of potential undefined values since they would always be require for the default -- typescript would take care of it.

I don't feel too strongly about keeping it if you think this seems confusing. This interface is removed after some refactoring, with a new interface for the configuration for Mapbox Static Image API url, and add in manual checking to ensure they are defined.